runtime/lua/vim/lsp/log.lua

```diff
@@ -31,21 +31,28 @@ do
     return logfilename
   end
 
-  vim.fn.mkdir(vim.fn.stdpath('cache'), "p")
-  local logfile = assert(io.open(logfilename, "a+"))
+  local logfile
+  ---@private
+  --- Opens log file.
+  local function open_logfile()
+    if logfile then return end
+    vim.fn.mkdir(vim.fn.stdpath('cache'), "p")
+    logfile = assert(io.open(logfilename, "a+"))
```

Comment:
Let's assume file can't be opened for some reason and the assert fails here. Each attempt at logging will try to open the file again, and most likely also fail. Should any further attempts at logging fail silently, or also throw an error?

Reply:
it should fail and raise an error once, if that's what you're asking